apps/ligo-virgo/webpack.config.js

```diff
@@ -181,6 +183,38 @@ module.exports = function (webpackConfig) {
                 publicPath: '/',
                 embedHeaderScript,
             }),
+        isProd &&
+            !isE2E &&
+            new CopyPlugin({
+                patterns: [
+                    {
+                        from: path.resolve(
+                            __dirname,
+                            './src/assets/manifest.json'
+                        ),
+                    },
+                    {
+                        from: path.resolve(
+                            __dirname,
+                            './src/assets/images/logo.png'
+                        ),
+                    },
+                ],
+            }),
+        isProd &&
+            !isE2E &&
+            new WorkboxPlugin.GenerateSW({
+                // these options encourage the ServiceWorkers to get in there fast
+                // and not allow any straggling "old" SWs to hang around
+                clientsClaim: true,
+                skipWaiting: true,
+                runtimeCaching: [
+                    {
+                        handler: 'NetworkFirst',
```

Comment:
I think it is better to use StaleWhileRevalidate for HTML and JS, and CacheFirst for image

Here is how I did it in my blog, just my opinion FYI https://github.com/linonetwo/wiki/blob/27608d25ac955dfb4670f63cdbf0da5ba837e6ec/public/service-worker.js

Reply:
Great suggestions. I will update.